### ISSUE TITLE
OptiFine's Reflector gets confused by multiple methods with the same name in dev environment

### ISSUE THREAD

**Issue by Barteks2x** [sp614x/optifine#917]
I used [this](https://github.com/octarine-noise/simpledeobf/) to deobfuscate OptiFine, and I found that there are some strange lighting issues.

After some investigation I found this in log
```
[12:21:03] [main/WARN] [Config/]: [OptiFine] (Reflector) More than one method found: net.minecraft.block.Block.getLightOpacity
[12:21:03] [main/WARN] [Config/]: [OptiFine] (Reflector)  - public int net.minecraft.block.Block.getLightOpacity(net.minecraft.block.state.IBlockState,net.minecraft.world.IBlockAccess,net.minecraft.util.math.BlockPos)
[12:21:03] [main/WARN] [Config/]: [OptiFine] (Reflector)  - public int net.minecraft.block.Block.getLightOpacity(net.minecraft.block.state.IBlockState)
[12:21:03] [main/WARN] [Config/]: [OptiFine] (Reflector) More than one method found: net.minecraft.block.Block.getLightValue
[12:21:03] [main/WARN] [Config/]: [OptiFine] (Reflector)  - public int net.minecraft.block.Block.getLightValue(net.minecraft.block.state.IBlockState)
[12:21:03] [main/WARN] [Config/]: [OptiFine] (Reflector)  - public int net.minecraft.block.Block.getLightValue(net.minecraft.block.state.IBlockState,net.minecraft.world.IBlockAccess,net.minecraft.util.math.BlockPos)
```

There is nothing the deobfuscator can do to fix it, this needs to be fixed by OptiFiine actually specifying method signature there.

**Comment by Barteks2x** [comment]
I was able to workaround the issue by using reflection to modify OptiFine's Reflector fields:
```java
        try {
            Class<?> reflectorClass = Class.forName("Reflector");
            Class<?> reflectorMethodClass = Class.forName("ReflectorMethod");
            Class<?> reflectorClassClass = Class.forName("ReflectorClass");

            Constructor<?> reflectorMethodConstructor = reflectorMethodClass.getConstructor(reflectorClassClass, String.class, Class[].class);

            Field ForgeBlock_field = reflectorClass.getField("ForgeBlock");
            Field ForgeBlock_getLightValue_field = reflectorClass.getField("ForgeBlock_getLightValue");
            Field ForgeBlock_getLightOpacity_field = reflectorClass.getField("ForgeBlock_getLightOpacity");

            Object ForgeBlock_value = ForgeBlock_field.get(null);

            Object ForgeBlock_getLightValue_newValue = reflectorMethodConstructor.newInstance(ForgeBlock_value, "getLightValue",
                    new Class[]{IBlockState.class, IBlockAccess.class, BlockPos.class});
            Object ForgeBlock_getLightOpacity_newValue = reflectorMethodConstructor.newInstance(ForgeBlock_value, "getLightOpacity",
                    new Class[]{IBlockState.class, IBlockAccess.class, BlockPos.class});

            ForgeBlock_getLightValue_field.set(null, ForgeBlock_getLightValue_newValue);
            ForgeBlock_getLightOpacity_field.set(null, ForgeBlock_getLightOpacity_newValue);
        } catch (ClassNotFoundException | NoSuchFieldException | IllegalAccessException | NoSuchMethodException | InstantiationException | InvocationTargetException e) {
            e.printStackTrace();
        }
```

**Comment by sp614x** [comment]
Fixed in preview C6 for 1.12.2
